Feature/nav 150 extract python viewer package

munterfi opened the pull request:
Still a draft, there are errors from the refactoring...

munterfi committed:
ORG: NAV-150 - Setup IDEA

munterfi committed:
ORG: NAV-150 - Copy vscode setup

munterfi committed:
ORG: NAV-150 - Setup poetry

munterfi committed:
DOC: NAV-150 - Update README

munterfi committed:
REFACTOR: NAV-150 - Copy viewer code

- Rename to snake_case
- Comment out new features for query params to be compatible with service version 0.4.0

munterfi committed:
TEST: NAV-150 - Add test folder

munterfi committed:
ORG: NAV-150 - Remove client

munterfi committed:
ORG: NAV-150 - Remove old viewer

munterfi committed:
ORG: NAV-150 - Remove flake8 configuration, since it is not set as development dependency in the project

- Could be added later, directly into the pyproject.toml:

[tool.flake8]
max-line-length = 88

- pre-commit can be configured later, at the moment it has to be bypassed all the time, since is quite a heavy refactoring.

munterfi committed:
ORG: NAV-150 - Enable dependabot

munterfi committed:
ORG: NAV-150 - Add poetry CI and Publish actions

munterfi committed:
ORG: NAV-150 - Add public-transit-client from PYPI

munterfi committed:
ORG: NAV-150 - Build info

munterfi committed:
ORG: NAV-150 - Add deploy script

- Python entrypoint, should work independent of directory location.
- Add watchdog, should increase performance, according to streamlit.

munterfi committed:
ORG: NAV-150 - Add dockerfile for building docker image

munterfi committed:
ORG: NAV-150 - Add Docker CI and Publish action

munterfi committed:
ORG: NAV-150 - Update version of client

munterfi committed:
DOC: NAV-150 - Update readme

munterfi committed:
ENH: NAV-150 - Get app up and running again

munterfi committed:
FIX: NAV-150 - Remove unit test marker from pytest in github action

munterfi committed:
TEST: NAV-150 - Add test which starts the app

clukas1 committed:
REFACTOR: NAV-150 - Use pathlib instead of os.

munterfi committed:
ORG: NAV-150 - Upgrade to service interface v0.5.0

munterfi committed:
ENH: NAV-150 - Enable service feature of v0.5.0

clukas1 committed:
ENH: NAV-150 - Check for all router features if query option is configurable.

clukas1 committed:
ENH: NAV-150 - Add schedule validity logic to date picker.

clukas1 committed:
STYLE: NAV-150 - Remove unused import.

clukas1 committed:
ORG: NAV-150 - Update client version to included type hinted version for mypy.

clukas1 committed:
ORG: NAV-150 - Add google docstring requirement.

clukas1 committed:
STYLE: NAV-150 - Remove some mypy type issues.

clukas1 committed:
FIX: NAV-150 - Fix incorrect keyword arguments.

clukas1 committed:
ORG: NAV-150 - Update poetry.lock file.

clukas1 committed:
ORG: NAV-150 - Add py.typed file.

clukas1 committed:
ORG: NAV-150 - Update version to 0.5.0

clukas1 committed:
STYLE: NAV-150 - Some pylint fixes.

munterfi committed:
ORG: NAV-150 - Remove TODOs since not possible

munterfi committed:
STYLE: NAV-150 - Format project and organize imports

munterfi committed:
ORG: NAV-150 - Update poetry lock

munterfi commented:
@clukas1, I think we could merge this and release the viewer with version 0.5.0?

munterfi committed:
DOC: NAV-150 - Update README.md

clukas1 commented on public_transit_viewer/pages/isolines.py:
```diff
@@ -0,0 +1,228 @@
+            popup = f"{data_row['targetStop']} - {data_row['durationFromSourceInMinutes']} min"
+
+        folium.Circle(  # type: ignore
+            location=source_coordinates,  # type: ignore
```
needs more logic:
`location=target_coordinates if time_type == TimeType.DEPARTURE else source_coordinates`

munterfi replied:
[b79fe3e](https://github.com/naviqore/public-transit-viewer/pull/14/commits/b79fe3ef86cc2de0103e8127595317a1e5d8b362)

clukas1 commented on public_transit_viewer/pages/isolines.py:
```diff
@@ -0,0 +1,228 @@
+        if filter_by == "connectionRound":
+            popup = f"{data_row['targetStop']} - Round {data_row['connectionRound']}"
+        else:
+            popup = f"{data_row['targetStop']} - {data_row['durationFromSourceInMinutes']} min"
```
same here, targetStop if timeType == DEPARTURE else sourceStop

munterfi replied:
[b79fe3e](https://github.com/naviqore/public-transit-viewer/pull/14/commits/b79fe3ef86cc2de0103e8127595317a1e5d8b362)

clukas1 commented on public_transit_viewer/pages/isolines.py:
```diff
@@ -0,0 +1,228 @@
+
+    # create a circle with the radius of the remaining distance
+    folium.Circle(  # type: ignore
+        location=[data_row["targetLat"], data_row["targetLon"]],  # type: ignore
```
also time type specific:
DEPARTURE == target else source

munterfi replied:
[b79fe3e](https://github.com/naviqore/public-transit-viewer/pull/14/commits/b79fe3ef86cc2de0103e8127595317a1e5d8b362)

clukas1 commented on public_transit_viewer/pages/isolines.py:
```diff
@@ -0,0 +1,228 @@
+        fill=True,
+        fill_color=color,
+        fill_opacity=0.3,
+        popup=f"{data_row['targetStop']} - {arrival_time} minutes",  # type: ignore
```
same logic issue

munterfi replied:
[b79fe3e](https://github.com/naviqore/public-transit-viewer/pull/14/commits/b79fe3ef86cc2de0103e8127595317a1e5d8b362)

clukas1 reviewed (changes requested):
@munterfi nice work. No real comments on your work. Just noticed a bug that marker points were not showing correctly depending on time_type. So think this could easily be fixed in this commit before we release 0.5.0

munterfi committed:
FIX: NAV-150 - Show marker points of isolines correctly depending on time_type

munterfi committed:
FIX: NAV-150 - Wait for static logo to load before starting app

- Avoids missing of logo.

munterfi committed:
REFACTOR: NAV-150 - Move utils to utils package

- Add __init__ files to sub packages.
- Add new asset package.
- Rename headers of pages.

munterfi committed:
ENH: NAV-150 - Rename main to connections

- Update page descriptions.
- Rename NAVIQORE_HOST_URL to NAVIQORE_SERVICE_URL.

munterfi committed:
FIX: NAV-150 - Remove sleep for logo loading, since it does not work always

munterfi committed:
ENH: NAV-150 - Add page favicon.ico
